### PR TITLE
0.2/177 user route is triggered if a user doesn t have a selected trader 2

### PR DESCRIPTION
**Pull request by charlesstrange2**
Also includes https://trello.com/c/ysCxzjqe/127-add-ui-element-to-make-clear-which-trader-working-on-behalf-of

### CODE REVIEW

File: src/components/Profile.vue [hunk 1]
```diff
@@ -2,13 +2,22 @@
...
 export default {
-    name: 'profile'
+    name: 'profile',
+    data() {
+        return {
+            selectedTrader: Store.trader
+        }
+    },
+    created : function() {
+        console.log(this.selectedTrader);
```
Comment (katjam) [inline src/components/Profile.vue]: Not saying anything.

File: src/main.js [hunk 2]
```diff
@@ -42,20 +42,30 @@ const router = new VueRouter({
 
 // Route Guard rules for directing users
```
Comment (katjam) [inline src/main.js:43]: Just to clarify for me... what's strangers vs friends only

File: src/pages/User.vue [hunk 3]
```diff
@@ -7,28 +7,20 @@
 
             <div class="content narrow">
 
-                <h1>Choose a trader account:</h1>
+                <form id="frmChooseTrader" v-on:submit.prevent >
 
-                <div class="form-group">
+                <h1>Choose a trader to manage</h1>
```
Comment (katjam) [inline src/pages/User.vue]: It feels like the page title should not be in the form.

I think form should only contain inputs, buttons etc - not other markup.

File: src/store.js [hunk 4]
```diff
@@ -1,11 +1,13 @@
...
-    user: {"id": 1},
+    user: {
+        id : 1,
+        traders : []
+    },
     trader: {
-        "id": 1,
-        pendedVouchers: []
+        id: null,
+        pendedVouchers: [],
```
Comment (katjam) [inline src/store.js]: remove , ?

**Review by katjam — CHANGES_REQUESTED** [review]
Just some small stuff.

File: src/main.js (hunk 2 above)
Comment (charlesstrange2) [reply src/main.js:43]: strangers = pages only logged out people can use
friends = pages that only logged in people can use

File: src/pages/User.vue (hunk 3 above)
Comment (charlesstrange2) [reply src/pages/User.vue]: Will move

File: src/store.js (hunk 4 above)
Comment (charlesstrange2) [reply src/store.js]: oke

File: src/main.js (hunk 2 above)
Comment (katjam) [reply src/main.js:43]: That's what I thought. Do we have strangers pages - besides login?

**Review by katjam — APPROVED** [review]
:ship: Once we get the tests passing. :+1:

**Pull request merged**